yazi-core/src/folder/stage.rs

```diff
@@ -1,7 +1,7 @@
-#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
 pub enum FolderStage {
 	#[default]
 	Loading,
 	Loaded,
-	Failed(std::io::ErrorKind),
+	Failed(std::io::ErrorKind, String),
```

Comment:
Why don't we generate error messages through `std::io::ErrorKind` instead of saving them as strings?

Reply:
How about using the [`Display` trait](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#impl-Display-for-ErrorKind) to do this?

Reply:
That doesn't give the pretty human readable messages, just the name of the kind.

Reply:
The error messages for the two cases `NotFound` and `PermissionDenied` are "entity not found" and "permission denied", which I think are quite readable. What does the current implementation look like?

Reply:
The error strings in the current commit are the classic C / POSIX error messages with the os error code, e.g. "No such file or directory (os error 2)" or "Permission denied (os error 13)". I wanted to use these since they seem familiar and are easy to search details for especially when you encounter a rarer type of error that you might not understand.

However, I guess the ErrorKind strings are actually good enough and it makes the implementation simpler.

Reply:
> However, I guess the ErrorKind strings are actually good enough and it makes the implementation simpler.

Nice, please change it! Let's implement it in the simplest way for now, if someone reports that it's not clear enough, we can create our own messages by `ErrorKind`.